src/cdk-experimental/scrollable-table-body/scrollable-table-body-layout.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, Inject, Injectable, Input} from '@angular/core';
+import {CdkTable} from '@angular/cdk/table/table';
+import {DOCUMENT} from '@angular/common';
+import {
+  _TABLE_LAYOUT_STRATEGY,
+  _TableLayoutStrategy,
+  _DefaultTableLayoutStrategy,
+} from '@angular/cdk/table/table-layout-strategy';
+
+/**
+ * A {@link _TableLayoutStrategy} that enables scrollable body content for flex tables.
+ */
+@Injectable()
+class ScrollableTableBodyLayoutStrategy implements _TableLayoutStrategy {
+  private defaultLayout: _DefaultTableLayoutStrategy;
+  private _pendingMaxHeight = 'none';
+  private _scrollViewport?: HTMLElement;
+  readonly headerCssClass = 'cdk-table-scrollable-table-header';
+  readonly bodyCssClass = 'cdk-table-scrollable-table-body';
+  readonly footerCssClass = 'cdk-table-scrollable-table-footer';
+
+  constructor(@Inject(DOCUMENT) private readonly _document: any) {
```

Comment:
Rather than making the injected `_document` a class member with `private`, you can create a separate property with type `Document` so that the types are correct for uses throughout the file.

(we have to inject it as any otherwise SSR breaks due to how TypeScript decorators output)